trusted-services.mk

```diff
@@ -53,3 +53,41 @@ ffa-sp-all: ffa-$1-sp
 ffa-sp-all-clean: ffa-$1-sp-clean
 ffa-sp-all-realclean: ffa-$1-sp-realclean
 endef
+
+################################################################################
+# Linux FF-A user space drivers
+################################################################################
+.PHONY: linux-arm-ffa-tee linux-arm-ffa-tee-clean
+all: linux-arm-ffa-tee
+
+linux-arm-ffa-tee: linux
+	mkdir -p $(OUT_PATH)/linux-arm-ffa-tee
+	$(MAKE) -C $(ROOT)/linux-arm-ffa-tee $(LINUX_COMMON_FLAGS) install \
+		TARGET_DIR=$(OUT_PATH)/linux-arm-ffa-tee
+
+linux-arm-ffa-tee-clean:
+	$(MAKE) -C $(ROOT)/linux-arm-ffa-tee clean
+
+# This driver is only used by the uefi-test app
+ifeq ($(TS_UEFI_TESTS),y)
+.PHONY: linux-arm-ffa-user linux-arm-ffa-user-clean
+all: linux-arm-ffa-user
+
+linux-arm-ffa-user: linux
+	mkdir -p $(OUT_PATH)/linux-arm-ffa-user
+	$(MAKE) -C $(ROOT)/linux-arm-ffa-user $(LINUX_COMMON_FLAGS) install \
+		TARGET_DIR=$(OUT_PATH)/linux-arm-ffa-user
+	echo "ed32d533-99e6-4209-9cc0-2d72cdd998a7" > \
+		$(OUT_PATH)/linux-arm-ffa-user/sp_uuid_list.txt
+
+linux-arm-ffa-user-clean:
+	$(MAKE) -C $(ROOT)/linux-arm-ffa-user clean
+
+# Disable CONFIG_STRICT_DEVMEM option in the Linux kernel config. This allows
+# userspace access to the whole NS physical address space through /dev/mem. It's
+# needed by the uefi-test app to communicate with the smm-gateway SP using a
+# static carveout.
+linux-defconfig:
```

Comment:
How does this work? Is the `linux-defconfig` target completely replaced with this or are commands just added?

Reply:
IMO the best option is to add a new config file, for example `kconfigs/fvp_trusted-services.conf`, and just do  `LINUX_DEFCONFIG_COMMON_FILES += kconfigs/fvp_trusted-services.conf`.

Reply:
Okay, done.